src/org/python/core/PyByteArray.java

```diff
@@ -1335,17 +1326,14 @@ public PyObject __iadd__(PyObject o) {
 
     @ExposedMethod(type = MethodType.BINARY, doc = BuiltinDocs.bytearray___iadd___doc)
     final synchronized PyObject bytearray___iadd__(PyObject o) {
-        PyType oType = o.getType();
-        if (oType == TYPE) {
-            // Use the general method, specifying the crack at the end of the array.
-            // Note this deals with the case o==this.
-            setslice(size, size, 1, (BaseBytes)o);
-        } else if (oType == PyString.TYPE) {
-            // Will fail if somehow not 8-bit clean
-            setslice(size, size, 1, (PyString)o);
```

Comment:
Because `PyString` is implemented using Java `char`, taking a buffer on it involves copying the whole thing to a `byte[]`, so the special `setslice` for `PyString` is worth it.